app/src/main/java/com/github/sdp/tarjetakuna/ui/singlecard/UsersFragment.kt

```diff
@@ -0,0 +1,126 @@
+package com.github.sdp.tarjetakuna.ui.singlecard
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.github.sdp.tarjetakuna.R
+import com.github.sdp.tarjetakuna.database.CardPossession
+import com.github.sdp.tarjetakuna.database.DBMagicCard
+import com.github.sdp.tarjetakuna.model.Coordinates
+import com.github.sdp.tarjetakuna.model.User
+import com.google.gson.Gson
+
+/**
+ * A fragment representing a list of [User].
+ */
+class UsersFragment : Fragment() {
+
+    private var ownedCards = true
+    private var dbMagicCard: DBMagicCard? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        arguments?.let {
+            ownedCards = it.getBoolean(ARG_IS_SHOWING_OWNED_CARDS)
+            dbMagicCard = try {
+                Gson().fromJson(it.getString(ARG_CARD), DBMagicCard::class.java)
+            } catch (e: Exception) {
+                null
+            }
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_user_list, container, false)
+
+        if (dbMagicCard == null) return view
+
+        val users = generateUsers().filter { user ->
+            user.cards.any { card ->
+                (card.possession == if (ownedCards) CardPossession.OWNED else CardPossession.WANTED)
+                        && (card.code == dbMagicCard!!.code)
+                        && (card.number == dbMagicCard!!.number)
+            } && user.email != currentUser().email
+        }.sortedBy { user ->
+            user.location.distanceKmTo(currentUser().location)
+        }
+
+        // Set the adapter
+        if (view is RecyclerView) {
+            with(view) {
+                layoutManager = LinearLayoutManager(context)
+                adapter = UserRecyclerViewAdapter(users, currentUser())
+            }
+        }
+        return view
+    }
+
+    //TODO: Replace / Remove when database is OK
```

Comment:
Will this be replaced or removed?

Reply:
It will depend on the database. But overall it is likely to be removed.